hypersync-client/src/from_arrow.rs

```diff
@@ -238,40 +289,42 @@ impl FromArrow for Log {
 }
 
 impl FromArrow for Trace {
-    fn from_arrow(batch: &ArrowBatch) -> Vec<Self> {
-        let from = batch.column::<BinaryArray<i32>>("from").ok();
-        let to = batch.column::<BinaryArray<i32>>("to").ok();
-        let call_type = batch.column::<Utf8Array<i32>>("call_type").ok();
-        let gas = batch.column::<BinaryArray<i32>>("gas").ok();
-        let input = batch.column::<BinaryArray<i32>>("input").ok();
-        let init = batch.column::<BinaryArray<i32>>("init").ok();
-        let value = batch.column::<BinaryArray<i32>>("value").ok();
-        let author = batch.column::<BinaryArray<i32>>("author").ok();
-        let reward_type = batch.column::<Utf8Array<i32>>("reward_type").ok();
-        let block_hash = batch.column::<BinaryArray<i32>>("block_hash").ok();
+    fn from_arrow_bin_array<B: List<Item = [u8]> + 'static, U: List<Item = str> + 'static>(
```

Comment:
The issue was that I want to be able to decode these types both from binary view arrays and simple binary arrays, but we only had the encoders (sorry, the from_arrow decoders) implemented for the simple binary arrays. I made it generic so that the `B` and the `U` parameters are respectively for the binary view arrays and the UTF-8 view arrays and their counterparts for simple binary.